packages/core/event.js

```diff
@@ -181,6 +193,19 @@ Event.create = function (maybeError, tolerateNonErrors, handledState, component,
   if (error.name === 'InvalidError') {
     event.addMetadata(`${component}`, 'non-error parameter', makeSerialisable(maybeError))
   }
+  if (error.cause) {
```

Comment:
nit: it seems mildly odd that you're checking for `error.cause` but passing `maybeError` to `getCauseStack`.  Perhaps this is the right thing to do, but just looks slightly surprising.

Reply:
that is surprising! it should have been `error`